scheduler/pkg/store/experiment/store.go

```diff
@@ -381,6 +381,7 @@ func (es *ExperimentStore) StopExperiment(experimentName string) error {
 			es.eventHub.PublishModelEvent(experimentStopEventSource, *modelEvt)
 		}
 		if pipelineEvt != nil {
+			// TODO: is this a bug?
```

Comment:
It is a bug... waiting to happen. At the moment we don't actually propagate the event source to the event handlers, because we don't take different actions depending on where the event is coming from. The only exception is for Servers, with the changes recently introduced [here](https://github.com/sakoush/seldon-core/blob/2f66806436e4ee82b81149824a20f660ad3464bc/scheduler/pkg/coordinator/server.go#L67)

As a consequence, setting the wrong source here has no effect right now, but we should update it nonetheless.

Reply:
Thanks for the context... updated it.